corehq/ex-submodules/casexml/apps/case/xml/generator.py

```diff
@@ -23,7 +24,13 @@ def safe_element(tag, text=None):
 
 
 def date_to_xml_string(date):
-    return json_format_date(date) if date else ''
+    if not date:
+        return ''
+
+    if isinstance(date, basestring):
+        date = parse_datetime(date)
```

Comment:
It's unlikely to error since the data is coming from ES which should be formatted correctly. If it does fail I think it would be better to fail hard so that we know about it.